### ISSUE TITLE
Query does not execute on 3.2.1 unless forcing Cypher 2.3

### ISSUE THREAD

**Issue by tomzeppenfeldt** [neo4j/neo4j#9550]
## Summary

We have a query, that generates an unexplainable error on 3.2.1.  The query runs well if we force CYPHER 2.3

The query looks like this (shortened version)

 ``` 
WITH 
        
        {
  name: "test for BIF. DO NOT DELETE !",
  lastUpdate: null,
  url: "www.testurl.org",
  sizeCategory: 2,
  activities: [ ],
  iatiType: 70,
  mainSubject: {
    name: "Debt forgiveness ",
    id: 20753
  },
  goal: "mygoal2",
  public: 1,
  privateComments: "some private comments",
  id: 28229,
  contactEmail: "test@gmail.com",
  start: "01-01-1966",
  comments: "my org comments"
}
        
        
        AS implementingOrg
		//find the :implementingOrg node and set all the fields
		MERGE (p:implementingOrg {id:implementingOrg.id})
		SET
			p.goal=implementingOrg.goal,
			p.start=implementingOrg.start,
			p.url=implementingOrg.url,
			p.comments=implementingOrg.comments,
			p.contactEmail=implementingOrg.contactEmail,
			p.privateComments= implementingOrg.privateComments,
			p.lastUpdate= implementingOrg.lastUpdate,
			p.updated = timestamp(),
			p.public= implementingOrg.public,
			p.iatiType = implementingOrg.iatiType,
			p.sizeCategory = implementingOrg.sizeCategory
		// remove all linked nodes at organisation
		WITH p,implementingOrg
		OPTIONAL MATCH (p)-[r1:mainSubject]->(n1)
		OPTIONAL MATCH (p)-[:activity]->(a)
		DELETE r1
     DETACH DELETE a

// cutoff point  ===========================================

 
WITH DISTINCT p,implementingOrg
		// add new links at organisation level
		//implementingOrg.mainSubject
		FOREACH (purpose IN implementingOrg.mainSubject |
		  MERGE (pp:Purpose {id:purpose.id})
		  CREATE (p)-[:mainSubject]->(pp)
		)
		//implementingOrg.activities
		FOREACH (a IN implementingOrg.activities |
			CREATE (pa:Activity {name:a.name,
								sequence:a.sequence,
								description:a.description,
								start:a.start,
								comments:a.comments
								}
			)
			SET pa.id=id(pa)
			CREATE (p)-[:activity]->(pa)
		)


``` 

When run in 3.2.1   , it throws  

```
Neo.DatabaseError.General.UnknownError
Unknown variable `r1`.
``` 

When cutting it off (at cutoff point) and only running the part until this point , it does NOT generate the error

When running it with  CYPHER 2.3. inserted , it executes perfectly.

**Comment by InverseFalcon** [comment]
We've seen that one before. I'll check for any relevant issue or pull request.

In the meantime, see if using the interpreted runtime works for you. Prefix query with:
`CYPHER runtime=interpreted `
and if the problem is still occurring, try forcing Cypher 3.1 by prefixing with:
`CYPHER 3.1 `

**Comment by tomzeppenfeldt** [comment]
Unfortunately

 `CYPHER runtime=interpreted`

 `CYPHER 3.1`

don't solve the issue.

**Comment by OliviaYtterbrink** [comment]
This is fixed by #9533 which will be in 3.2.2
